docs/src/extend/custom-rule-tutorial.md

```diff
@@ -0,0 +1,452 @@
+                                    return fixer.replaceText(node.declaration.init, "bar");
+                                }
+                            });
+                        }
+                    }
+                }
+            }
+        };
+    }
+};
+```
+
+## Step 5: Set up Testing
+
+With the rule written, you can test it to make sure it's working as expected.
+
+ESLint provides the built-in [`RuleTester`](../integrate/nodejs-api#ruletester) class to test rules. You do not need to use third-party testing libraries to test ESLint rules, but `RuleTester` works seamlessly with tools like Mocha and Jest.
+
+Next, create the file for the tests, `enforce-foo-bar.test.js`:
+
+```shell
+touch enforce-foo-bar.test.js
+```
+
+And add a test script to your `package.json` file to run the tests:
+
+```javascript
+// package.json
+{
+    // ...other configuration
+    "scripts": {
+        "test": "node enforce-foo-bar.test.js"
+    },
+    // ...other configuration
+}
+```
+
+## Step 6: Write the Test
+
+To write the test using `RuleTester`, import the class and your custom rule into the `enforce-foo-bar.test.js` file.
+
+The `RuleTester#run()` method tests the rule against valid and invalid test cases. If the rule fails to pass any of the test scenarios, this method throws an error.
+`RuleTester` requires that at least one valid and one invalid test scenarios be present.
+
+```javascript
+// enforce-foo-bar.test.js
+const {RuleTester} = require("eslint");
+const fooBarRule = require("./enforce-foo-bar");
+
+const ruleTester = new RuleTester({
+  // Must use at least ecmaVersion 2015 because
+  // that's when `const` variable introduced.
+  parserOptions: { ecmaVersion: 2015 }
+});
+
+// Throws error if the tests in ruleTester.run() do not pass
+ruleTester.run(
+  "foo-bar", // rule name
+  fooBarRule, // rule code
+  { // checks
+    // 'valid' checks cases that should pass
+    valid: [{
+      code: "const foo = 'bar';",
+    }],
+    // 'invalid' checks cases that should not pass
+    invalid: [{
+      code:"const foo = 'baz';",
+      output: 'const foo = "bar";',
+      errors: 1,
+    }],
+  }
+);
+
+console.log("All tests passed!");
+```
+
+## Step 7: Bundle the Custom Rule in a Plugin
+
+Now that you've written the custom rule and validated that it works, you can include it in a plugin. Using a plugin, you can share the rule in an npm package to use in other projects.
+
+Create the file for the plugin:
+
+```shell
+touch eslint-plugin-example.js
+```
+
+And now write the plugin code. Plugins are just exported JavaScript objects. To include a rule in a plugin, include it in the plugin's `rules` object, which contains key-value pairs of rule names and their source code.
+
+To learn more about creating plugins, refer to [Create Plugins](plugins).
+
+```javascript
+// eslint-plugin-example.js
+
+const fooBarRule = require("./enforce-foo-bar");
+const plugin = { rules: { "foo-bar": fooBarRule } };
```

Comment:
I think the rule name should be the same as the file name, either both `enforce-foo-bar` or both `foo-bar`.

Reply:
done